python/cudf/cudf/testing/testing.py

```diff
@@ -0,0 +1,581 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+from typing import Union
+
+import numpy as np
+import pandas as pd
+
+import cudf
+from cudf.utils.dtypes import is_categorical_dtype
+
+
+def _check_isinstance(left, right, obj):
+    if not isinstance(left, obj):
+        raise AssertionError(
+            f"{obj} Expected type {obj}, found {type(left)} instead"
+        )
+    elif not isinstance(right, obj):
+        raise AssertionError(
+            f"{obj} Expected type {obj}, found {type(right)} instead"
+        )
+
+
+def raise_assert_detail(obj, message, left, right, diff=None):
+
+    msg = f"""{obj} are different
+
+{message}
+[left]:  {left}
+[right]: {right}"""
+
+    if diff is not None:
+        msg += f"\n[diff]: {diff}"
+
+    raise AssertionError(msg)
+
+
+def _check_types(
+    left, right, check_categorical=True, exact="equiv", obj="Index"
+):
+    if not exact or exact == "equiv":
+        if (
+            isinstance(left, cudf.RangeIndex)
+            and isinstance(right, cudf.Int64Index)
```

Comment:
Unlike Pandas, we have `Int8Index`, `Int16Index`, etc., and we probably also want to include those here.